Sources/AppHost/OAAppDelegate.mm

```diff
@@ -189,37 +189,44 @@ - (BOOL) application:(UIApplication *)application openURL:(NSURL *)url sourceApp
     else if ([scheme isEqualToString:@"osmandmaps"])
     {
         NSDictionary *params = [OAUtilities parseUrlQuery:url];
-        
-        // osmandmaps://?lat=45.6313&lon=34.9955&z=8&title=New+York
-        double lat = [params[@"lat"] doubleValue];
-        double lon = [params[@"lon"] doubleValue];
-        double zoom = [params[@"z"] doubleValue];
-        NSString *title = params[@"title"];
-        
-        Point31 pos31 = [OANativeUtilities convertFromPointI:OsmAnd::Utilities::convertLatLonTo31(OsmAnd::LatLon(lat, lon))];
-        dispatch_async(dispatch_get_main_queue(), ^{
+        if (params.count != 0){
+            // osmandmaps://?lat=45.6313&lon=34.9955&z=8&title=New+York
+            double lat = [params[@"lat"] doubleValue];
+            double lon = [params[@"lon"] doubleValue];
+            double zoom = [params[@"z"] doubleValue];
+            NSString *title = params[@"title"];
+            NSString *navigate = [url host];
             
-            OAMapViewController* mapViewController = [_rootViewController.mapPanel mapViewController];
-            
-            UIViewController *top = _rootViewController.navigationController.topViewController;
-            
-            if (![top isKindOfClass:[JASidePanelController class]])
-                [_rootViewController.navigationController popToRootViewControllerAnimated:NO];
-
-            if (_rootViewController.state != JASidePanelCenterVisible)
-                [_rootViewController showCenterPanelAnimated:NO];
-
-            [_rootViewController.mapPanel closeDashboard];
-            
-            [mapViewController goToPosition:pos31 andZoom:zoom animated:NO];
+            Point31 pos31 = [OANativeUtilities convertFromPointI:OsmAnd::Utilities::convertLatLonTo31(OsmAnd::LatLon(lat, lon))];
+            dispatch_async(dispatch_get_main_queue(), ^{
+                
+                OAMapViewController* mapViewController = [_rootViewController.mapPanel mapViewController];
+                
+                UIViewController *top = _rootViewController.navigationController.topViewController;
+                
+                if (![top isKindOfClass:[JASidePanelController class]])
+                    [_rootViewController.navigationController popToRootViewControllerAnimated:NO];
+                
+                if (_rootViewController.state != JASidePanelCenterVisible)
+                    [_rootViewController showCenterPanelAnimated:NO];
+                
+                [_rootViewController.mapPanel closeDashboard];
+                
+                [mapViewController goToPosition:pos31 andZoom:zoom animated:NO];
+                OATargetPoint *targetPoint = [mapViewController.mapLayers.contextMenuLayer getUnknownTargetPoint:lat longitude:lon];
+                if (title.length > 0)
+                    targetPoint.title = title;
+                if ([navigate  isEqual: @"navigate"]){
```

Comment:
What if we already navigating? Did you test it?

Reply:
Yes, so the app pops up a menu to add as an additional location or new route.  I think thats what we would want?

Reply:
yes, ok